frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+class Root extends React.Component {
+  constructor() {
+    super()
```

Comment:
Here, you're overriding the constructor for the parent class `React.Component`. When you call super, you're calling the original `constructor` function that is defined for that parent class.

The problem is that the parent class actually takes an argument. So, you should really be doing this:

```javascript
constructor(props) {
  super(props);
```

Or, if you aren't going to be using props, you should probably just pass the arguments through:

```javascript
constructor(...args) {
  super(...args);
```